tests/e2e/util.go

```diff
@@ -16,3 +42,198 @@ func ReadModuleTemplate(filepath string) (*v1beta2.ModuleTemplate, error) {
+	var deployment appsv1.Deployment
+	err := k8sClient.Get(ctx, client.ObjectKey{
+		Namespace: namespace,
+		Name:      deploymentName,
+	}, &deployment)
+
+	GinkgoWriter.Println("Available replicas:", deployment.Status.AvailableReplicas)
+	return err == nil && deployment.Status.AvailableReplicas != 0
+}
+
+func KymaCRIsInReadyState(ctx context.Context,
+	k8sClient client.Client,
+	kymaName string,
+	namespace string) bool {
+	var kyma v1beta2.Kyma
+	err := k8sClient.Get(ctx, client.ObjectKey{
+		Namespace: namespace,
+		Name:      kymaName,
+	}, &kyma)
+
+	return err == nil && kyma.Status.State == v1beta2.StateReady
+}
+
+func ApplyModuleTemplate(
+	moduleTemplatePath string) error {
+	cmd := exec.Command("kubectl", "apply", "-f", moduleTemplatePath)
+
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("%w: %v", errModuleTemplateNotApplied, err)
+	}
+
+	return nil
+}
+
+func enableKymaModuleWithExpectedExitCode(moduleName string, expectedExitCode int) error {
+	cmd := exec.Command("kyma", "alpha", "enable", "module", moduleName, "-w")
+	err := cmd.Run()
+	var exitCode int
+	if exitErr, ok := err.(*exec.ExitError); ok {
+		exitCode = exitErr.ExitCode()
+	} else {
+		exitCode = cmd.ProcessState.ExitCode()
+	}
+
+	GinkgoWriter.Println("Exit code", exitCode)
+	if exitCode != expectedExitCode {
+		return fmt.Errorf("%w: %v", errModuleEnablingFailed, err)
+	}
+	return nil
+}
+
+func EnableKymaModuleWithReadyState(moduleName string) error {
+	return enableKymaModuleWithExpectedExitCode(moduleName, exitCodeNoError)
+}
+
+func EnableKymaModuleWithWarningState(moduleName string) error {
+	return enableKymaModuleWithExpectedExitCode(moduleName, exitCodeWarning)
+
+}
+
+func DisableModuleOnKyma(moduleName string) error {
+	cmd := exec.Command("kyma", "alpha", "disable", "module", moduleName, "-w")
+	err := cmd.Run()
+	var exitCode int
+	if exitErr, ok := err.(*exec.ExitError); ok {
+		exitCode = exitErr.ExitCode()
+	} else {
+		exitCode = cmd.ProcessState.ExitCode()
+	}
+
+	GinkgoWriter.Println("Exit code", exitCode)
+	if exitCode != 0 {
+		return fmt.Errorf("%w: %v", errModuleDisablingFailed, err)
+	}
+	return nil
+}
+
+func CRDIsAvailable(ctx context.Context,
+	k8sClient client.Client,
+	name string) bool {
+	var crd v1extensions.CustomResourceDefinition
+	err := k8sClient.Get(ctx, client.ObjectKey{
+		Name: name,
+	}, &crd)
+
+	return err == nil
+}
+
+func crIsInExpectedState(resourceType string,
+	resourceName string,
+	namespace string,
+	expectedState string) bool {
+	cmd := exec.Command("kubectl", "get", resourceType, resourceName, "-n",
+		namespace, "-o", "jsonpath='{.status.state}'")
+
+	statusOutput, err := cmd.CombinedOutput()
+	GinkgoWriter.Println(string(statusOutput))
+
+	return err == nil && string(statusOutput) == expectedState
+}
+
+func CRIsReady(resourceType string,
+	resourceName string,
+	namespace string) bool {
+	return crIsInExpectedState(resourceType, resourceName, namespace, crReadyState)
```

Comment:
you should make this function accept `v1beta2.State` instead of redefine a const for state